chat_shell/chat_shell/tools/builtin/knowledge_listing.py

```
@@ -430,10 +433,34 @@ async def _arun(
+                    successful_doc_ids = [
+                        d.get("id") for d in docs if "error" not in d and d.get("id")
+                    ]
+                    # Persist for each knowledge base
+                    for kb_id in self.knowledge_base_ids:
+                        await self._persist_kb_head_result(
+                            kb_id, successful_doc_ids, total_chars
+                        )
+                except (json.JSONDecodeError, KeyError) as e:
+                    logger.warning(f"[KbHeadTool] Failed to parse result for persistence: {e}")
+
+            return result
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)